In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/house-prices-advanced-regression-techniques/sample_submission.csv
/kaggle/input/house-prices-advanced-regression-techniques/data_description.txt
/kaggle/input/house-prices-advanced-regression-techniques/train.csv
/kaggle/input/house-prices-advanced-regression-techniques/test.csv


In [2]:
# 파일 받기, Id -> Index로 사용

df_train = pd.read_csv('../input/house-prices-advanced-regression-techniques/test.csv', index_col='Id')
df_test = pd.read_csv('../input/house-prices-advanced-regression-techniques/test.csv', index_col='Id')

In [3]:
df_train.columns

Index(['MSSubClass', 'MSZoning', 'LotFrontage', 'LotArea', 'Street', 'Alley',
       'LotShape', 'LandContour', 'Utilities', 'LotConfig', 'LandSlope',
       'Neighborhood', 'Condition1', 'Condition2', 'BldgType', 'HouseStyle',
       'OverallQual', 'OverallCond', 'YearBuilt', 'YearRemodAdd', 'RoofStyle',
       'RoofMatl', 'Exterior1st', 'Exterior2nd', 'MasVnrType', 'MasVnrArea',
       'ExterQual', 'ExterCond', 'Foundation', 'BsmtQual', 'BsmtCond',
       'BsmtExposure', 'BsmtFinType1', 'BsmtFinSF1', 'BsmtFinType2',
       'BsmtFinSF2', 'BsmtUnfSF', 'TotalBsmtSF', 'Heating', 'HeatingQC',
       'CentralAir', 'Electrical', '1stFlrSF', '2ndFlrSF', 'LowQualFinSF',
       'GrLivArea', 'BsmtFullBath', 'BsmtHalfBath', 'FullBath', 'HalfBath',
       'BedroomAbvGr', 'KitchenAbvGr', 'KitchenQual', 'TotRmsAbvGrd',
       'Functional', 'Fireplaces', 'FireplaceQu', 'GarageType', 'GarageYrBlt',
       'GarageFinish', 'GarageCars', 'GarageArea', 'GarageQual', 'GarageCond',
       'PavedDrive', 'Wo

In [4]:
df_train.describe()

MSSubClass  LotFrontage       LotArea  OverallQual  OverallCond  \
count  1459.000000  1232.000000   1459.000000  1459.000000  1459.000000   
mean     57.378341    68.580357   9819.161069     6.078821     5.553804   
std      42.746880    22.376841   4955.517327     1.436812     1.113740   
min      20.000000    21.000000   1470.000000     1.000000     1.000000   
25%      20.000000    58.000000   7391.000000     5.000000     5.000000   
50%      50.000000    67.000000   9399.000000     6.000000     5.000000   
75%      70.000000    80.000000  11517.500000     7.000000     6.000000   
max     190.000000   200.000000  56600.000000    10.000000     9.000000   

         YearBuilt  YearRemodAdd   MasVnrArea   BsmtFinSF1   BsmtFinSF2  ...  \
count  1459.000000   1459.000000  1444.000000  1458.000000  1458.000000  ...   
mean   1971.357779   1983.662783   100.709141   439.203704    52.619342  ...   
std      30.390071     21.130467   177.625900   455.268042   176.753926  ...   
min    1879.000000   1950.000000     0.000000     0.000000     0.000000  ...   
25%    1953.000000   1963.000000     0.000000     0.000000     0.000000  ...   
50%    1973.000000   1992.000000     0.000000   350.500000     0.000000  ...   
75%    2001.000000   2004.000000   164.000000   753.500000     0.000000  ...   
max    2010.000000   2010.000000  1290.000000  4010.000000  1526.000000  ...   

        GarageArea   WoodDeckSF  OpenPorchSF  EnclosedPorch    3SsnPorch  \
count  1458.000000  1459.000000  1459.000000    1459.000000  1459.000000   
mean    472.768861    93.174777    48.313914      24.243317     1.794380   
std     217.048611   127.744882    68.883364      67.227765    20.207842   
min       0.000000     0.000000     0.000000       0.000000     0.000000   
25%     318.000000     0.000000     0.000000       0.000000     0.000000   
50%     480.000000     0.000000    28.000000       0.000000     0.000000   
75%     576.000000   168.000000    72.000000       0.000000     0.000000   
max    1488.000000  1424.000000   742.000000    1012.000000   360.000000   

       ScreenPorch     PoolArea       MiscVal       MoSold       YrSold  
count  1459.000000  1459.000000   1459.000000  1459.000000  1459.000000  
mean     17.064428     1.744345     58.167923     6.104181  2007.769705  
std      56.609763    30.491646    630.806978     2.722432     1.301740  
min       0.000000     0.000000      0.000000     1.000000  2006.000000  
25%       0.000000     0.000000      0.000000     4.000000  2007.000000  
50%       0.000000     0.000000      0.000000     6.000000  2008.000000  
75%       0.000000     0.000000      0.000000     8.000000  2009.000000  
max     576.000000   800.000000  17000.000000    12.000000  2010.000000  

[8 rows x 36 columns]

In [5]:
df_train.head()

MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
Id                                                                      
1461          20       RH         80.0    11622   Pave   NaN      Reg   
1462          20       RL         81.0    14267   Pave   NaN      IR1   
1463          60       RL         74.0    13830   Pave   NaN      IR1   
1464          60       RL         78.0     9978   Pave   NaN      IR1   
1465         120       RL         43.0     5005   Pave   NaN      IR1   

     LandContour Utilities LotConfig  ... ScreenPorch PoolArea PoolQC  Fence  \
Id                                    ...                                      
1461         Lvl    AllPub    Inside  ...         120        0    NaN  MnPrv   
1462         Lvl    AllPub    Corner  ...           0        0    NaN    NaN   
1463         Lvl    AllPub    Inside  ...           0        0    NaN  MnPrv   
1464         Lvl    AllPub    Inside  ...           0        0    NaN    NaN   
1465         HLS    AllPub    Inside  ...         144        0    NaN    NaN   

     MiscFeature MiscVal  MoSold  YrSold  SaleType  SaleCondition  
Id                                                                 
1461         NaN       0       6    2010        WD         Normal  
1462        Gar2   12500       6    2010        WD         Normal  
1463         NaN       0       3    2010        WD         Normal  
1464         NaN       0       6    2010        WD         Normal  
1465         NaN       0       1    2010        WD         Normal  

[5 rows x 79 columns]

In [6]:
df_train.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1459 entries, 1461 to 2919
Data columns (total 79 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   MSSubClass     1459 non-null   int64  
 1   MSZoning       1455 non-null   object 
 2   LotFrontage    1232 non-null   float64
 3   LotArea        1459 non-null   int64  
 4   Street         1459 non-null   object 
 5   Alley          107 non-null    object 
 6   LotShape       1459 non-null   object 
 7   LandContour    1459 non-null   object 
 8   Utilities      1457 non-null   object 
 9   LotConfig      1459 non-null   object 
 10  LandSlope      1459 non-null   object 
 11  Neighborhood   1459 non-null   object 
 12  Condition1     1459 non-null   object 
 13  Condition2     1459 non-null   object 
 14  BldgType       1459 non-null   object 
 15  HouseStyle     1459 non-null   object 
 16  OverallQual    1459 non-null   int64  
 17  OverallCond    1459 non-null   int64  
 18  YearB

In [7]:
df_train.loc[:, 'Street']

Id
1461    Pave
1462    Pave
1463    Pave
1464    Pave
1465    Pave
        ... 
2915    Pave
2916    Pave
2917    Pave
2918    Pave
2919    Pave
Name: Street, Length: 1459, dtype: object

In [8]:
sum(df_train.loc[:, 'Street'] == 'Pave')

1453

>* 'Street'에서는 1459개 중 1453개가 모두 'Pave'이다.
>* 큰 영향을 미치지 않을 거 같다.

In [9]:
df_train.drop('Street', axis=1)
df_test.drop('Street', axis=1)

MSSubClass MSZoning  LotFrontage  LotArea Alley LotShape LandContour  \
Id                                                                           
1461          20       RH         80.0    11622   NaN      Reg         Lvl   
1462          20       RL         81.0    14267   NaN      IR1         Lvl   
1463          60       RL         74.0    13830   NaN      IR1         Lvl   
1464          60       RL         78.0     9978   NaN      IR1         Lvl   
1465         120       RL         43.0     5005   NaN      IR1         HLS   
...          ...      ...          ...      ...   ...      ...         ...   
2915         160       RM         21.0     1936   NaN      Reg         Lvl   
2916         160       RM         21.0     1894   NaN      Reg         Lvl   
2917          20       RL        160.0    20000   NaN      Reg         Lvl   
2918          85       RL         62.0    10441   NaN      Reg         Lvl   
2919          60       RL         74.0     9627   NaN      Reg         Lvl   

     Utilities LotConfig LandSlope  ... ScreenPorch PoolArea PoolQC  Fence  \
Id                                  ...                                      
1461    AllPub    Inside       Gtl  ...         120        0    NaN  MnPrv   
1462    AllPub    Corner       Gtl  ...           0        0    NaN    NaN   
1463    AllPub    Inside       Gtl  ...           0        0    NaN  MnPrv   
1464    AllPub    Inside       Gtl  ...           0        0    NaN    NaN   
1465    AllPub    Inside       Gtl  ...         144        0    NaN    NaN   
...        ...       ...       ...  ...         ...      ...    ...    ...   
2915    AllPub    Inside       Gtl  ...           0        0    NaN    NaN   
2916    AllPub    Inside       Gtl  ...           0        0    NaN    NaN   
2917    AllPub    Inside       Gtl  ...           0        0    NaN    NaN   
2918    AllPub    Inside       Gtl  ...           0        0    NaN  MnPrv   
2919    AllPub    Inside       Mod  ...           0        0    NaN    NaN   

     MiscFeature  MiscVal  MoSold  YrSold  SaleType SaleCondition  
Id                                                                 
1461         NaN        0       6    2010        WD        Normal  
1462        Gar2    12500       6    2010        WD        Normal  
1463         NaN        0       3    2010        WD        Normal  
1464         NaN        0       6    2010        WD        Normal  
1465         NaN        0       1    2010        WD        Normal  
...          ...      ...     ...     ...       ...           ...  
2915         NaN        0       6    2006        WD        Normal  
2916         NaN        0       4    2006        WD       Abnorml  
2917         NaN        0       9    2006        WD       Abnorml  
2918        Shed      700       7    2006        WD        Normal  
2919         NaN        0      11    2006        WD        Normal  

[1459 rows x 78 columns]